ratelimit/http.go

```diff
@@ -119,15 +138,60 @@ func (l rateLimiter) Handler(next http.Handler) http.Handler {
 		w.Header().Set(headerKeyReset, strconv.FormatInt(resetTime, 10))
 
 		if err != nil {
-			if err == errOverLimit {
+			// in all cases, where we will return an error
+			// if the rate limiter becomes overloaded or unavailable, we should start to force the client to back off
+			if retryAfter, ok := l.getAndSetNextRetryAfterValue(); ok {
+				w.Header().Set(headerRetryAfter, retryAfter)
+			}
+
+			if errors.Is(err, errOverLimit) {
```

Comment:
In this case shouldn't we set the `headerRetryAfter` header to `resetTime`?

Reply:
I mean `resetTime-now()`

Reply:
yeah looks like I missed the reset. So maybe it makes sense on the happy path to just delete the key, wdyt? That would reset us to min value?